play/src/front/Phaser/Game/GameMap/GameMapFrontWrapper.ts

```diff
@@ -75,6 +75,7 @@ export class GameMapFrontWrapper {
      */
     private mapChangedSubject = new Subject<number[][]>();
     private areaUpdatedSubject = new Subject<AreaData>();
+    private entitiesReadySubject = new Subject<boolean>();
```

Comment:
This `entitiesReadySubject` is a bit overkill in my opinion.

First, it will resolve only once, so what you want is a Promise, not a Subject (a subject can trigger several times, a promise only once)

Then, we have that only because you cannot await a constructor. So we are probably doing too much in the constructor.

You should instead take the async content of the constructor and put it in a `init(): Promise<void>` method that we can call just after the `GameMapFrontWrapper` is instanciated.

Then, you can simply put a ".then()" on the promise returned by the ".init()" method.

Reply:
I see the point of that but I would rather stick to the current approach of keeping GameMapFrontWrapper initialization within the constructor (as we instantiate it in two possible places, thus we would have to call init() twice too).
We can switch from subscribing to use `toPromise` on observable though, skipping subscription / unsubscribing entirely, like this:

```
.getEntitiesReadyAsObservable().toPromise().then((ready) => {...}
```